Eshop.WebApi/Controllers/CategoryController.cs

```diff
@@ -16,37 +24,38 @@ public class CategoryController : Controller
         [HttpGet]
         public List<Category> GetCategories()
         {
-            return Categories;
+            return dbContext.Categories.ToList(); ;
         }
 
         [HttpGet("{id}")]
         public Category GetCategory(int id) 
         { 
-            return Categories.First(x => x.Id == id);
+            return dbContext.Categories.First(x => x.Id == id);
         }
 
         [HttpPost]
         public Category CreateCategory(string name) 
         { 
```

Comment:
If you change this code to this, Create will also work. With a 0 id you delegate to the persistence layer to take care of it.
Db is having autoincrement in it by definition in CategoryConfiguration.cs HasKey()
![image](https://github.com/user-attachments/assets/b9e94e44-fc7c-402f-8219-662a6338671e)

Reply:
interesting, I added that